ci/Jenkinsfile-databricks

```diff
@@ -0,0 +1,246 @@
+    while(!ready) {
+        def json = sh(script: "databricks clusters get --cluster-id ${clusterId}", returnStdout: true).trim()
+        def jsonSlurper = new JsonSlurperClassic()
+        def cfg = jsonSlurper.parseText(json)
+        ready = cfg["state"] == "RUNNING"
+        Thread.sleep(10000)
+    }
+}
+
+def deleteCluster(clusterId) {
+    sh "databricks clusters delete --cluster-id ${clusterId}"
+}
+
+static def getDatabricksSparkVersions(props) {
+    def sparkVersions = props["supportedSparkVersions"].split(" ").toList()
+    def boundaryVersion = props["databricksTestSinceSpark"]
+    def list = new ArrayList<String>()
+    list.addAll(sparkVersions.subList(sparkVersions.indexOf(boundaryVersion), sparkVersions.size()))
+    return list
+}
+
+def waitForArtifactToAttach(clusterId, artifact, type) {
+    def ready = false
+    while(!ready) {
+        def rawJson = sh(script: "databricks libraries cluster-status --cluster-id ${clusterId}", returnStdout: true).trim()
+        def parsedJson = new JsonSlurperClassic().parseText(rawJson)
+        def status = parsedJson.library_statuses.find { it.library[type] == "dbfs:/${clusterId}/${artifact}" }.status
+        ready = status == "INSTALLED"
+        Thread.sleep(1000)
+    }
+}
+
+def installDependencies(clusterId, scalaArtifact, pythonArtifact, rArtifact) {
+    // Scala
+    sh "databricks fs cp assembly/build/libs/${scalaArtifact} dbfs:/${clusterId}/${scalaArtifact}"
+    sh "databricks libraries install --cluster-id ${clusterId} --jar dbfs:/${clusterId}/${scalaArtifact}"
+    waitForArtifactToAttach(clusterId, scalaArtifact, "jar")
+    // Python
+    sh "databricks fs cp py/build/pkg/dist/${pythonArtifact} dbfs:/${clusterId}/${pythonArtifact}"
+    sh "databricks libraries install --cluster-id ${clusterId} --whl dbfs:/${clusterId}/${pythonArtifact}"
+    waitForArtifactToAttach(clusterId, pythonArtifact, "whl")
+    // R
+    sh "databricks fs cp r/build/${rArtifact} dbfs:/FileStore/${clusterId}/${rArtifact}"
+    sh "databricks fs cp h2o-3/h2o-r/h2o_*.99999.tar.gz dbfs:/FileStore/${clusterId}/h2o.tar.gz"
+}
+
+def test(workspaceDir, clusterId, language) {
+    def extension = language
+    if (language == "python") {
+        extension = "py"
+    }
+    sh "databricks workspace import --language ${language} ci/databricksTests/test.${extension} ${workspaceDir}/test.${extension}"
+    def rawJson = sh(script: """
+        databricks runs submit --json '
+            {
+              "name": "Test ${language}",
+              "existing_cluster_id": "${clusterId}",
+              "notebook_task": {
+                "notebook_path": "${workspaceDir}/test.${extension}"
+              }
+            }
+            '
+        """, returnStdout: true).trim()
+    def parsedJson = new JsonSlurperClassic().parseText(rawJson)
+    def runId = parsedJson.run_id
+    waitForJobToFinish(runId, language)
+}
+
+def waitForJobToFinish(jobId, language) {
+    def done = false
+    while(!done) {
```

Comment:
Is a progress reported anywhere?

Reply:
You can see progress in Jenkins console

Reply:
So each loop iteration prints out something to stdout?

Reply:
yes

Reply:
ok, thanks!